### DISCUSSION TITLE
Unable to Store Downloaded Content in SD Card

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#710] [General Discussion]
Hello Team,

We are using release 4.6 apk.
I am encountering an issue with storing downloaded content in the SD card while using the 4.6 release APK. In the application, there is a feature to download content. When I navigate to the download tab and attempt to change the storage setting to SD storage, I encounter a problem. Upon clicking on the SD storage option, a permission popup appears. After granting permission, the radio button shifts back to Phone storage instead of staying on SD storage.

Steps to Reproduce:
1. Open the application.
2. Navigate to the download tab.
3. Click on the option to change storage setting.
4. Choose SD storage option.
5. Grant permission when prompted.
6. Observe that the radio button switches back to Phone storage instead of staying on SD storage.

Attempts Made:
Debugging: I have attempted to debug the issue but have not identified any apparent problem.Here is the screen short for the same.
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/5527844c-fb03-4433-8fa9-63ebe6d8192b)

Request for Guidance:
I seek guidance on how to utilize this functionality effectively. Despite my attempts to debug, I have been unable to resolve the issue. Any assistance on troubleshooting or resolving this matter would be greatly appreciated.

Thank you for your attention to this matter.

cc: @rajeevsathish @mohitga @vinukumar-vs @ashokreddy1208 @swayangjit

**Comment by mohitga** [comment]
@swayangjit can you guide @siddhishinde0723  on this issue?

**Reply by siddhishinde0723** [reply]
@swayangjit As discussed, I have updated the following permission but am still facing the issue:

Permission: 
`  <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />`

Issue:
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/93600cfe-599c-4010-b541-a56d95981507)

**Reply by siddhishinde0723** [reply]
@swayangjit I added the this two permission and action still facing the issue. 

1. Declare the [MANAGE_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#MANAGE_EXTERNAL_STORAGE) permission in the manifest.
2. Use the [ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION](https://developer.android.com/reference/android/provider/Settings#ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION) intent action to direct users to a system settings page where they can enable the following option for your app: Allow access to manage all files.

**Reply by swayangjit** [reply]
@siddhishinde0723 This is existing issue in the current release. The fix will be available in the upcoming releases and I will update in this thread once fix is available.
